app/services/provider_interface/candidate_withdrawal_data_by_provider.rb

```diff
@@ -0,0 +1,51 @@
+module ProviderInterface
+  class CandidateWithdrawalDataByProvider
+    CONFIG_PATH = 'config/withdrawal_reasons.yml'.freeze
+
+    attr_reader :provider
+
+    def initialize(provider:)
+      @provider = provider
+    end
+
+    def submitted_withdrawal_reason_count
+      ApplicationChoice
+        .where('provider_ids @> ARRAY[?]::bigint[]', provider)
+        .where.not(structured_withdrawal_reasons: [])
+        .where(current_recruitment_cycle_year: RecruitmentCycle.current_year)
+        .count
```

Comment:
is there any re-use to be had with `withdrawal_query` below?

Reply:
🆗